internal/repos/modules.go

```diff
@@ -35,6 +35,35 @@ type module struct {
 	Main          bool
 }
 
+var regexMixedVersioning = regexp.MustCompile(`^(.*?)-([0-9]{14})-([a-fA-F0-9]{12})$`)
+
+func toRepoRule(mod module) Repo {
+	var tag, commit string
+	if strings.HasPrefix(mod.Version, "v0.0.0-") {
+		i := strings.LastIndex(mod.Version, "-")
```

Comment:
Please note that "strings.LastIndex()" **cannot** fail as we already know from line 42 that the prefix contains at least one dash.